src/main/java/ee/tuleva/onboarding/comparisons/fundvalue/retrieval/GlobalStockIndexRetriever.java

```diff
@@ -0,0 +1,187 @@
+        ArrayList<DailyRecord> records = new ArrayList<>();
+
+        FtpClient ftpClient = new FtpClient(ftpHost, ftpUsername, ftpPassword, ftpPort);
+        try {
+            ftpClient.open();
+            Collection<String> fileNames = ftpClient.listFiles(GlobalStockIndexRetriever.PATH);
+            for(LocalDate date = startDate; date.isBefore(endDate) || date.isEqual(endDate); date = date.plusDays(1))
+            {
+                String dateString = date.format(DateTimeFormatter.ofPattern("yyyyMMdd"));
+                String fileName = fileNames.stream()
+                    .filter(string -> string.endsWith(dateString + ".zip"))
+                    .findAny()
+                    .orElse(null);
+
+                if(fileName == null) {
+                    continue;
+                }
+
+                try{
+                    InputStream fileStream = ftpClient.downloadFileStream(GlobalStockIndexRetriever.PATH + fileName);
+                    Optional<DailyRecord> optionalRecord = findInZip(fileStream, GlobalStockIndexRetriever.secID);
+                    optionalRecord.ifPresent(dailyRecord -> pushDailyRecord(records, dailyRecord));
+                    fileStream.close();
+                    ftpClient.completePendingCommand();
+                } catch(IOException ignored) {}
+            }
+        } catch(IOException e) {
+            log.error(e.getMessage());
+        } finally {
+            try {
+                ftpClient.close();
+            } catch(IOException ignored) {}
+        }
+        return extractValuesFromRecords(records, startDate, endDate);
+    }
+
+    private List<FundValue> extractValuesFromRecords(List<DailyRecord> records, LocalDate startDate, LocalDate endDate) {
+        ArrayList<FundValue> fundValues = new ArrayList<>();
+
+        for(LocalDate date = startDate; date.isBefore(endDate) || date.isEqual(endDate); date = date.plusDays(1)) {
+            String monthID = date.format(DateTimeFormatter.ofPattern("yyyyMM"));
+            int dayOfMonth = date.getDayOfMonth();
+
+            DailyRecord record = records.stream()
+                .filter(r -> r.secID.equals(secID) && r.monthID.equals(monthID))
+                .findAny()
+                .orElse(null);
+
+            if(record == null)
+                continue;
+
+            String value = record.values.get(dayOfMonth - 1);
+            if(!value.isEmpty()) {
+                fundValues.add(new FundValue(GlobalStockIndexRetriever.KEY, date, new BigDecimal(value)));
+            }
+        }
+        return fundValues;
+    }
+
+    private void pushDailyRecord(List<DailyRecord> records, DailyRecord record) {
+        int recordIdx = records.indexOf(record);
+        if(recordIdx != -1) {
+            records.get(recordIdx).update(record);
+        } else {
+            records.add(record);
+        }
+    }
+
+    private Optional<DailyRecord> findInZip(InputStream stream, String secId) throws IOException {
```

Comment:
What is `secId`? `securityId`? Always prefer longer, more descriptive, variable names.